datalad_registry/tasks.py

```diff
@@ -16,6 +18,31 @@
 
 InfoType = Dict[str, Union[str, float]]
 
+def clone_dataset(url: str, ds_path: Path) -> Any:
+    import datalad.api as dl
+
+    ds_path_str = str(ds_path)
+    # TODO (later): Decide how to handle subdatasets.
+    # TODO: support multiple URLs/remotes per dataset
+    # Make remote name to be a hash of url. Check below if among
+    # remotes and add one if missing, then use that remote, not 'origin'
+    if ds_path.exists():
+        ds = dl.Dataset(ds_path)
+        ds_repo = ds.repo
+        ds_repo.fetch(all_=True)
+        ds_repo.call_git(["reset", "--hard", "refs/remotes/origin/HEAD"])
+    else:
+        ds = dl.clone(url, ds_path_str)
+    return ds
+
+def get_info(ds_repo: Any) -> InfoType:
```

Comment:
```suggestion
def get_info(ds_repo: "GitRepo") -> InfoType:
```
? or if class must be loaded, then add `from datalad.support.gitrepo import GitRepo` among imports

Reply:
Nevermind; it fails when testing with `--devserver`.  Quoting `"GitRepo"` works, though it'll cause problems if flake8 is added (which I hope to add alongside black once there are no outstanding PRs).

Reply:
Nevermind again; mypy doesn't like it when `"GitRepo"` quoted without importing the name (and I'm now going to add type checking to tox.ini).

Reply:
> Nevermind; it fails when testing with `--devserver`

heh heh -- asyncio strikes back (according to that original comment).  Should not be an issue after we release datalad 0.15.0, but I guess no need to worry about that ATM.